lib/kubernetes-deploy/kubernetes_resource/base_set.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class BaseSet < KubernetesResource
+    def deploy_failed?
+      @pods.present? && @pods.any?(&:deploy_failed?)
```

Comment:
To make a point related to this specific method: I'd rather sick with  `.all?` for ReplicaSet. For DaemonSet, `.any?` makes sense because the pods are more unique in that a bad node can cause a single one to fail, and that one will not be able to be rescheduled. One bad pod on a RS, where they're truly clones, is much more likely to be recovered automatically. For StatefulSet I could go either way but would lean toward `any?` because the nature of SS suggests the pods could fail for independent, unrecoverable reasons. IMO this decision needs to be made class-by-class, balancing deploy fragility/speed concerns, and the base class shouldn't define a default as a result.